src/component/Nav/Nav.js

```diff
@@ -1,8 +1,88 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import Modal from './component/Modal';
 import './Nav.scss';
 
 const Nav = () => {
-  return <div>Nav</div>;
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const updateScroll = () => {
+    setScrollPosition(window.scrollY || document.documentElement.scrollTop);
+  };
+
+  useEffect(() => {
+    window.addEventListener('scroll', updateScroll);
+  });
+
+  const openModal = () => {
+    setModalOpen(true);
+  };
+  const closeModal = () => {
+    setModalOpen(false);
+  };
+
+  const scrollNavHandle = () => {
+    return scrollPosition < 5 ? '' : ' changeNav';
+  };
+
+  const scrollLinktagHandle = () => {
+    return scrollPosition < 5 ? '' : ' changeLinktagText';
+  };
+
+  const scrollimgHandle = () => {
+    return scrollPosition < 5 ? '' : ' changeImg';
+  };
```

Comment:
```suggestion

  const isScrollMoved = scrollPosition < 5;
  const switchLinkColor = isScrollMoved ? '' : ' changeLinktagText';
  const switchNavBackgroundColor = isScrollMoved ? '' : ' changeNav';
  const switchImgColor = isScrollMoved ? '' : ' changeImg';
```

반복되는 조건을 변수화 해서 반복되는 부분에 변수를 넣어주는건 어떨까요
함수를 생성하고 호출하기보다, 변수로 참조하는게 다루기 간편해 보입니다.